src/ol/events/eventtarget.js

```diff
@@ -64,12 +71,17 @@ ol.events.EventTarget.prototype.dispatchEvent = function(event) {
   evt.target = this;
   var listeners = this.listeners_[type];
   if (listeners) {
-    for (var i = listeners.length - 1; i >= 0; --i) {
-      if (listeners[i].call(this, evt) === false ||
-          evt.propagationStopped) {
+    this.pendingRemovals_[type] = 0;
+    for (var i = 0, ii = listeners.length; i < ii; ++i) {
+      if (listeners[i].call(this, evt) === false || evt.propagationStopped) {
```

Comment:
It would be nice to clean up the pending removals here too.  Maybe set a flag and `break`.  Then `return false` after cleaning up if the flag is set.